server/core/src/finalize_utilities.cpp

```diff
@@ -270,4 +263,32 @@ namespace irods
 
         return ec;
     } // close_replica_without_catalog_update
+
+    auto close_replica_and_unlock_data_object(RsComm& _comm, const int _fd, const int _status) -> int
+    {
+        const auto [replica, replica_lm] = ir::duplicate_replica(*L1desc[_fd].dataObjInfo);
+        const auto remove_rst_entry      = irods::at_scope_exit{[data_id = replica.data_id()] { rst::erase(data_id); }};
+
+        constexpr auto preserve_rst = true;
+        const auto close_ec = irods::close_replica_without_catalog_update(_comm, _fd, preserve_rst);
+        if (close_ec < 0) {
+            irods::log(LOG_ERROR, fmt::format(
+                "[{}:{}] - Failed to close replica "
+                "[error_code=[{}], path=[{}], hierarchy=[{}]]",
+                __FUNCTION__, __LINE__, close_ec,
+                replica.logical_path(), replica.hierarchy()));
+        }
+
+        if (const int ec = ill::unlock_and_publish(_comm, replica.data_id(), replica.replica_number(), _status, ill::restore_status); ec < 0) {
+            irods::log(LOG_ERROR, fmt::format(
+                "[{}:{}] - Failed to unlock data object on failure to create physical file "
```

Comment:
we know this was during a 'create physical file'?

Reply:
I think this is a copy-pasta. Changed to "Failed to unlock data object"